cardinal/ecs/world.go

```diff
@@ -307,17 +308,26 @@ func (w *World) ListMessages() ([]message.Message, error) {
 
 // NewWorld creates a new world.
 func NewWorld(
-	nonceStore storage.NonceStorage,
+	schemaAndNonceStore redis.EngineStorage,
 	entityStore store.IManager,
 	namespace Namespace,
 	opts ...Option,
 ) (*World, error) {
 	logger := &ecslog.Logger{
 		&log.Logger,
 	}
+	nonceStore, ok := schemaAndNonceStore.(redis.NonceStorage)
```

Comment:
These type conversions are unnecessary. The program would fail to compile if schemaAndNonceStore didn't implement these interfaces.

Lines 329 and 330 below can just be:

nonceStore: schemaAndNonceStore
schemaStore: schemaAndNonceStore

Reply:
This change wasn't done, because scott requested a change in the architecture.